### ISSUE TITLE
Getting Stuck at Welcome Screen

### ISSUE THREAD

**Issue by CS-JS** [FreeRDP/FreeRDP#3429]
I have an issue, that seems to occur quite randomly. When I log in to different machines (Windows 7 x64, Windows Server 2008) sometimes, xfreerdp gets stuck at the Windows "Welcome Screen". As a test, I closed the xfreerdp window after a few minutes without any reaction.

My connection string is pretty straightforward:
`jschoen@NB-CS-JS:~$ xfreerdp /v:*IP address* /u:Admin`

I also tried different options like -compression (same issue), /compression-level:0 (same issue), as i read here about #3231 . This happens with authentication on network as well as when I set the connection security to RDP. When set to RDP the login screen works fine, I select my user and type the password but get stuck afterwards at the following "welcome" screen.

The session itself seems to be connected fine, when I check the logged on users in task manager in the console session, I am displayed as a connected user via rdp correctly.

I first experienced this in Remmina, my favourite RDP Session Manager on Linux, but it is also reproducable using the connection string above directly from commandline.

My xfreerdp version is:

```
jschoen@NB-CS-JS:~$ xfreerdp  --version
This is FreeRDP version 2.0.0-dev (git n/a)
```

While xfreerdp is stuck at the welcome screen, I dont have any console output - is there any way to increase the debug level or anything? :) If I can provide additional information to this issue, let me know.

Best regards

**Comment by bmiklautz** [comment]
@CS-JS on which distribution are you on? Can you reproduce this also with our nightly builds? https://github.com/FreeRDP/FreeRDP/wiki/PreBuilds

**Comment by CS-JS** [comment]
Kubuntu 16.04,
I installed the freerdp-nightly package, but autocomplete only suggests xfreerdp, which still is 
```
jschoen@NB-CS-JS:~$ xfreerdp  --version
This is FreeRDP version 2.0.0-dev (git n/a)
```

How do I confirm that I run the nightlies? Sorry for the question...

**Comment by bmiklautz** [comment]
@CS-JS the binaries from the nightly package are installed into /opt/freerdp-nightly/bin/ to not conflict with the distribution native package. So to test run
```
 /opt/freerdp-nightly/bin/xfreerdp
```

**Comment by CS-JS** [comment]
Thanks, I could not reproduce the error in my last connections with the nightly builds.
On the other side, this issue only occured from time to time - so when I run into the problem again, I will switch to the nightlys and try them.

Maybe we can put this issue on hold until then? Thanks so far :)

**Comment by Cheb57** [comment]
Hello,

I have a got the similar problem as described on this discussion thread. Note: i'm using a reverse ssh socks proxy to access from home to an office laptop running on Win7 x64 Enterprise edition. It is a first try to connect rdp using FreeRDP under Linux (Debian x86). I built FreeRDP from sources, using stable-1.1 and master branches. Login by using both revisions can not continue more than the Welcome stuck message. My build is compiled with debugging symbols and differents debug flags during the cmake project generation. So i can help to understand the problem. Also i have to the Win7 laptop a VNC access thanks to the socks remote port from home to office (to check event viewer and log account eventually).
Master branch revision used: xfreerdp --version
This is FreeRDP version 2.0.0-dev (git a132545)

Thanks!

**Comment by Cheb57** [comment]
The problem i had seems to be related to a win7 server issue. The error log off received is sent by the server even if i'm already disconnected (or session locked) using my account from host. Unfortunately, at home i only have a windows 8 basic edition without RDP server. So exept at work no other possibility to test. Anyway, compiled the win32 bin wfreerdp is working at work by using the same target win7 x64 laptop when connected from an other computer inside the corporate network. On my side i will not add more comments/posts on this discussion thread. Eventually i will open a new one for my use. Thank you.

**Comment by Cheb57** [comment]
The issue on my side was due to the the Win7 RDP server side implementation, using ssh tunneling and using the loopback localhost address. In this case, win7 drops immediatelly the connection during the Welcome message after logon (and stuck until timeout). Tunneling ports (remote port forwarding) from home to a second office laptop and use RDP is working fine by example. Then connect to the target laptop is also working from the second laptop.

Win7 is dropping connection client from localhost address. Using others ports more than 3390 did not solve the problem. I need to use a second office laptop as gateway to target the desired laptop. So the FreeRDP client was not the problem.

**Comment by jbrockerville** [comment]
This is old, but I wanted to post this somewhere. I also could not get an RDP connection to work over a reverse SSH tunnel. This is gonna sound wonky, but Windows 7 apparently does not like allowing remote logins to the user that initiated the tunnel. I read that somewhere and figured what the heck I got nothing else to lose, but to my surprise it worked immediately. On PC-A, target Win7 machine, I created a dummy user, logged in with the dummy user, started a reverse tunnel (with PuTTY) to PC B, and switched back to the user I wanted. Then, on PC-B, connected to PC-A via RDP and it got past the hanging Welcome screen and subsequent timeout on the first try. I hope someone finds this helpful!
